android/src/main/java/com/theweflex/react/WeChatModule.java

```diff
@@ -156,31 +206,161 @@ public void sendAuthRequest(String scope, String state, Callback callback) {
+                }
+            });
+        } else {
+            this.sendShareRequest(media, null, data, callback);
         }
-        _share(SendMessageToWX.Req.WXSceneTimeline, data, callback);
     }
 
-    @ReactMethod
-    public void shareToSession(ReadableMap data, Callback callback) {
-        if (api == null) {
-            callback.invoke(NOT_REGISTERED);
-            return;
+    private void sendShareRequest(WXMediaMessage.IMediaObject media, Bitmap thumb, ReadableMap data, Callback callback) {
+        WXMediaMessage message = new WXMediaMessage();
+        message.mediaObject = media;
+        if (data.hasKey("title")) {
+            message.title = data.getString("title");
+        }
+        if (data.hasKey("description")) {
+            message.description = data.getString("description");
         }
-        _share(SendMessageToWX.Req.WXSceneSession, data, callback);
+        if (data.hasKey("mediaTagName")) {
+            message.mediaTagName = data.getString("mediaTagName");
+        }
+        if (data.hasKey("messageAction")) {
+            message.messageAction = data.getString("messageAction");
+        }
+        if (data.hasKey("messageExt")) {
+            message.messageExt = data.getString("messageExt");
+        }
+        if (thumb != null) {
+            if (thumb.length() / 1024 > THUMB_SIZE) {
+                message.thumbData = bitmapResizeGetBytes(thumb, THUMB_SIZE);
+            } else {
+                message.thumbData = thumb;
+            }
+        }
+
+        SendMessageToWX.Req req = new SendMessageToWX.Req();
+        req.scene = data.hasKey("scene") ? data.getInt("scene") : SendMessageToWX.Req.WXSceneSession;
+        req.transaction = UUID.randomUUID().toString();
+        req.message = message;
+        callback.invoke(null, api.sendReq(req));
     }
 
+    /**
+     * Share the plain text
+     * @param data
+     * @param callback
+     */
     @ReactMethod
-    public void shareToFavorite(ReadableMap data, Callback callback) {
-        if (api == null) {
-            callback.invoke(NOT_REGISTERED);
+    public void shareText(ReadableMap data, Callback callback) {
+        WXTextObject media = new WXTextObject();
+        media.text = data.getString("text");
+        this.sendShareRequest(media, null, data, callback);
+    }
+
+    /**
+     * 分享图片
+     * @param data
+     * @param callback
+     */
+    @ReactMethod
+    public void shareImage(final ReadableMap data, final Callback callback) {
+        Uri imgUrl;
+        try {
+            imgUrl = Uri.parse(data.getString("imageUrl"));
+            if (imgUrl.getScheme() == null) {
+                // handle static resource if no schema is provided.
+                imgUrl = getResourceDrawableURI(getReactApplicationContext(), imgUrl);
+            }
+        } catch (Exception ex) {
+            imgUrl = null;
+        }
+
+        if (imgUrl == null) {
+            callback.invoke(null);
```

Comment:
```suggestion
            callback.invoke(new Object[]{});
```
@yorkie 这里对于非 varargs 的调用, 使用 Object[], 不然编译报错。